### ISSUE TITLE
Bad performance in Keras: It diverges if sample_weight is not added to train_in_batch but sample_weight_mode="temporal" is in Model.compile. 

### ISSUE THREAD

**Issue by jpviguerasguillen** [tensorflow/tensorflow#34233]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes.
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Google Colab, with the latest packages.
- TensorFlow installed from (source or binary): Provided by Google Colab.
- TensorFlow version (use command below): 2.0.0
- Python version: 3.7.x

**Describe the current behavior**
This might not be a bug, but a change has happened in the recent days in the Keras API that has led my models to either divergence to NaNs or reduced performance. Let me explain (long explanation):

For the last 3 months, I have been training a Dense U-net to solve the problem from 'Digest Challenge- Conference MICCAI 2019':  **https://digestpath2019.grand-challenge.org/Home/**
I have done many experiments, using Google Colab and Tensorflow 2.0 with integrated Keras. I never specify the version of the packages, so I use whatever Google Colab loads by default. And for months that was perfectly fine. But one week ago, on 7th November 2019, something was changed in the Keras API that all my models went crazy.

To give some context, this dataset contains histopathology images, i.e. images of the human colon where cancer cells are present. The dataset is divided into two parts: Negative images, where all tissue is healthy, and Positive images, where cancer tissue is present. Thus, there are two classes of pixels: cancer and non-cancer. The goal is to do segmentation of the cancer tissue. The graph below show the accuracy in the training set (in blue) and in the validation set, where the Positive images (in red) and Negative images (in green) are displayed separately. Usually, the 12 non-stop hours that Google Colab provides allows for 40 iterations. So the following day I load the weights and keep training.

Since these images are huge, I take patches, build my own batch (trying to balance the classes), and use Keras function _train_on_batch_ for training. Sometimes, I have used the option of _sample_weight_ from _train_on_batch_ to weight the classes even further or simply to play with the weights ...

```   tLoss = model.train_on_batch(batch_data, batch_label, sample_weight=batch_weight)```
adding the option sample_weight_mode="temporal" in compile() ...
```    
model.compile(optimizer=nadam, 
             sample_weight_mode="temporal",
             loss= 'binary_crossentropy', 
             metrics=['binary_accuracy'])
```
So far so good!
Now, in the past, I have sometimes forgotten to remove the line _sample_weight_mode="temporal"_ when I was **NOT** using _sample_weight_ in _train_on_batch_, yet there were NO changes in the performance (this is, either removing or not that line in _compile()_ gave the same performance if no actual weighting was introduced in _train_on_batch_ ... or at least I do not remember observing any difference, but I cannot ensure this statement). 

This was like that until a week ago! Suddenly, if I am NOT using _sample_weight_ but I have the line _sample_weight_mode="temporal"_ in compile(), the network diverges to a fixed loss-value and an accuracy of 0.5 to both classes (sometimes it is not straight forward... maybe after some iterations).

Then I discovered that I had to remove that line to avoid this. And that makes sense, so it would not be an issue. The problem is that, suddenly, the performance has been reduced a lot (in the Positive images, red line). In the following graph, you could see how the accuracy was going quite well, reaching a plateau in both types of images (Positives and Negatives), until one day that I load the model to keep training and this situation happens (on iteration 175). Since then, the network cannot reach the performance that was obtained before.

![Accuracy_Validation2019 11 09](https://user-images.githubusercontent.com/52449827/68773542-0b87d380-062c-11ea-8c61-8d468eca45b7.png)

So, to summarize, I was training using _train_on_batch_ with NO input argument _sample_weight_ but WITH the line _sample_weight_mode="temporal"_ in compile() (from iteration 0 until iteration 175 in the graph). It went well. At one point in time, that setting made the network to diverge (and I highlight that I did NOT change my code whatsoever, simply load the network and keep training). Once I removed the line _sample_weight_mode="temporal"_ in compile(), the network did not diverge but performance decreased (from iteration 175 forward in the graph).
I have replicated this in other old networks (and in new networks trained from scratch) and the same situation happens.

Is this a bug? I am not sure!
My hypotheses: 
- Some kind of weighting was internally done before, which was corrected recently, leading to a decrease in my current performance. In this case, I would be interested in knowing what was changed.
- It is a new bug introduced recently.

**Code to reproduce the issue**
See Jupyter Notebook: https://drive.google.com/open?id=1gMnrIe3WBRyEpGumisMMP8ZqqTm5XtX1
The link to the dataset is within the Notebook.

**Comment by gowthamkpr** [comment]
@jpviguerasguillen Can you please provide a colab notebook reproducing this issue. Thanks!

**Comment by jpviguerasguillen** [comment]
The Jupyter notebook to replicate the issue is:
https://drive.google.com/open?id=1gMnrIe3WBRyEpGumisMMP8ZqqTm5XtX1

**Comment by jpviguerasguillen** [comment]
**UPDATE:**
I am trying tf_nightly (tensorflow version 2.1.0) and the divergence issue seems to be solved (the training metrics look very stable, as it used to be).
Therefore, I am closing this thread.
Thanks!
